### PR TITLE
Media Gallery component now receives allowable file types from File Uploader

### LINKED ISSUE #23881: "Select from Gallery" option on media file uploader does not use correct allowable file types
### Preconditions (*)
1. Magento 2.4-develop vanilla

### Steps to reproduce (*)

1. Content -> Configuration
2. Global -> Edit
3. HTML Head -> Favicon icon
4. Select from Gallery
5. Choose Files
6. Select "<anyfile>.ico" from local filesystem

### Expected result (*)
1. icon file is successfully uploaded

### Actual result (*)
1. Error "We don't recognize or support this file extension type." is reported
Peek 2020-08-31 13-22

### Investigation
Allowed file types is stated "JPG, GIF, PNG, ICO, APNG" on design config webpage
Open Browser Dev tools and monitor network traffic when hitting "Select from Gallery" button.
Inspect inital ajax request's response back from server

```
allowedExtensions = 'jpg jpeg png gif';
acceptFileTypes: /(\.|\/)(gif|jpe?g|png)$/i,

```
May be fixed https://github.com/magento/magento2/pull/25007

### PR DESCRIPTION
**Pull request by gwharton**
### Description (*)
Extended allowable extensions for image type to include ico and apng (These are listed for the Favicon)

Magento Ui image uploader component now passes the list of allowable file types to the Media Gallery component. This will (optionally) override the default list of allowable file types obtained from di.xml. Previously the Media Gallery would just use the default set of allowable types as set in di.xml, even if the Image Uploader component was configured for a different set.

Implemented fallback thumbnail placeholder generation for Media Gallery when filetype is not supported by current Image Adapter. As no ico thumbnails can be generated by GD2, it shows the placeholder image instead when GD2 Image Adapter is selected.

Added a check to the uploader to skip resizing uploaded image for thumbnail generation if the image file extension is ico and the Image Adapter is GD2. 

Corrected error message style.

Added MFTF test.

### Related Pull Requests
Replaces the following PR
1. Closes #25007: ICO file not supported in media file uploader fixed, issue #23881

### Fixed Issues (if relevant)
1. Closes #23881: "Select from Gallery" option on media file uploader does not use correct allowable file types

### Manual testing scenarios (*)
As per original issue

### Questions or comments
Although ico is added to the list of allowable file types, and you are able to upload them, they are NOT supported by the GD2 Image Adapter which is the default. Therefore additional logic was required to skip thumbnail generation for the uploaded image, which although was handled gracefully on upload, caused an error message to be shown on the Media Gallery screen when ico files were uploaded. The error message is no longer shown.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by gwharton** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @gwharton. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @gwharton, here is your [new Magento instance](https://pr-26980.instances.magento-community.engineering).
Admin access: [https://pr-26980.instances.magento-community.engineering/admin_7f58](https://pr-26980.instances.magento-community.engineering/admin_7f58)
Login: `eda376e7` Password: `648283ed150d`
Instance will be terminated in up to 3 hours.

**Comment by gwharton** [comment]
For discussion, yes we are displaying a warning saying that the image is not supported, but actually, in terms of favicon, the image format is very much supported. It is only unsupported in terms of generating a thumbnail for the media browser.

The problem is, this is just the generic error message spewed out by the GD2 image adapter when it receives an unsupported image.

**Comment by gwharton** [comment]
Functional Tests currently failing due to lack of ImageMagick php module in CI environment. Will investigate if this can be added, otherwise, I will remove the ImageMagic Image Adapter based tests.

File: app/code/Magento/Cms/Block/Adminhtml/Wysiwyg/Images/Content/Uploader.php [hunk 1]
```diff
@@ -71,4 +71,18 @@ protected function _getMediaType()
...
+    /**
+     * Return allowed file extensions based on request or config
+     *
+     * @return array
+     */
+    protected function _getAllowedFileExtensions()
+    {
+        if ($encodedExtensions = $this->getRequest()->getParam('allowed_extensions')) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            return(explode(" ", base64_decode($encodedExtensions)));
```
Comment (lbajsarowicz) [inline app/code/Magento/Cms/Block/Adminhtml/Wysiwyg/Images/Content/Uploader.php:84]: @piotrekkaminski This does not look safe for me. Could you take a look?

Comment (gwharton) [reply app/code/Magento/Cms/Block/Adminhtml/Wysiwyg/Images/Content/Uploader.php:84]: What would you recommend. I couldn't think of a better way of passing the list of file types into the media gallery component than this. Open to suggestions.

**Pull request closed without merging**